benchexec/tools/metaval.py

```diff
@@ -0,0 +1,94 @@
+"""
+BenchExec is a framework for reliable benchmarking.
+This file is part of BenchExec.
+
+Copyright (C) 2007-2019  Dirk Beyer
+All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+import argparse
+import benchexec.util as util
+import benchexec.tools.template
+import benchexec.result as result
+import os
+import threading
+
+
+class Tool(benchexec.tools.template.BaseTool):
+
+    TOOL_TO_PATH_MAP = {"cpachecker": "CPAchecker", "esbmc": "esbmc"}
+    REQUIRED_PATHS = list(TOOL_TO_PATH_MAP.values())
+
+    def __init__(self):
+        self.lock = threading.Lock()
+
+    def executable(self):
+        return util.find_executable("metaval.sh")
+
+    def name(self):
+        return "metaval"
+
+    def determine_result(self, returncode, returnsignal, output, isTimeout):
+        if not hasattr(self, "wrappedTool"):
+            return "METAVAL ERROR"
+        return self.wrappedTool.determine_result(
+            returncode, returnsignal, output, isTimeout
+        )
+
+    def cmdline(self, executable, options, tasks, propertyfile=None, rlimits={}):
+        parser = argparse.ArgumentParser(add_help=False, usage=argparse.SUPPRESS)
+        parser.add_argument("--witness", required=True)
+        parser.add_argument("--metaval", required=True)
+        (knownargs, options) = parser.parse_known_args(options)
+        verifierName = knownargs.metaval.lower()
+        witnessName = knownargs.witness
+        with self.lock:
+            if not hasattr(self, "wrappedTool"):
+                self.wrappedTool = __import__(
+                    "benchexec.tools." + verifierName, fromlist=["Tool"]
+                ).Tool()
+            else:
+                if not verifierName == self.wrappedTool.name().tolower():
```

Comment:
There is no guarantee that the name of the tool-info module equals `Tool.name().tolower()`. Indeed, it is already not always the case, for example for Ultimate-based tools. The `name()` function is supposed to return a human-readable name, which includes spaces in many cases.

Either store the module name, or try `inspect.getmodule()` on the tool instance or its class.

Reply:
it is easier to store the name, done